docs/proposals/formatted-output.md

```diff
@@ -379,6 +379,105 @@ oras discover $REGISTRY/$REPO:v1 --format json
 > [!NOTE]
 > The `--format` flag will replace the existing `--output` flag. The `--output` will be marked as "deprecated" in ORAS v1.2.0 and will be removed in the future releases.
 
+If the referrers have associated referrers, ORAS should be able to show the manifest content of the subject image and all referrers. 
+
+When showing the subject image and all referrers' manifests recursively in a tree view output, the following fields should be returned:
+
+- `reference`: full reference by digest of the subject image
+- `mediaType`: media type of the subject image
+- `digest`: digest of the subject image
+- `size`: subject image size in bytes
+- `manifests`: the list of referrers' manifest
+  - `reference`: full reference by digest of the referrer
+  - `mediaType`: media type of the referrer
+  - `digest`: digest of the referrer
+  - `size`: referrer file size in bytes
+  - `artifactType`: artifact type of a referrer
+  - `annotations`: contains arbitrary metadata in a referrer
+  - `referrerManifests`: the list of referrers' manifest
+
+For example, when there are two refferers lifecycle metadata and in-toto attestation associated with a sample image, the signatures are associated with these two files respectively. The output in a tree view will be:
+
+```bash
+oras discover localhost:5000/kubernetes/kubectl@sha256:bece4f4746a39cb39e38451c70fa5a1e5ea4fa20d4cca40136b51d9557918b01
+```
+
+```console
+localhost:5000/kubernetes/kubectl@sha256:bece4f4746a39cb39e38451c70fa5a1e5ea4fa20d4cca40136b51d9557918b01
+├── application/vnd.oci.artifact.lifecycle
+│   └── sha256:325129be79f416fe11a9ec44233cfa57f5d89434e6d37170f97e48f7904983e3
+│       └── application/vnd.cncf.notary.signature
+│           └── sha256:f520330e9f43c05859c532e67a25c9c765b144782ae7b872656192c27fd4e2dd
+└── application/vnd.in-toto+json
+    └── sha256:a811606b09341bab4bbc0a4deb2c0cb709ec9702635cbe2d36b77d58359ec046
+        └── application/vnd.cncf.notary.signature
+            └── sha256:04723fd7d00df77c6f226b907667396554bf9418dc48a7a04feb5ff24aa0b9ec
+```
+
+When showing the subject image and all referrers' manifests recursively in a pretty JSON output, the following JSON output should be returned:
+
+```bash
+oras discover localhost:5000/kubernetes/kubectl@sha256:bece4f4746a39cb39e38451c70fa5a1e5ea4fa20d4cca40136b51d9557918b01 --format json --recursive
+```
+
+```
```

Comment:
nit: this code block is not a valid json

Reply:
Added JSON code block